crates/pathfinder/src/state/sync.rs

```diff
@@ -55,40 +35,48 @@ pub fn sync(
         .connection()
         .context("Creating database connection")?;
 
-    let (tx_events, mut rx_events) = mpsc::channel(1);
+    let (tx_l1, mut rx_l1) = mpsc::channel(1);
```

Comment:
2 queues because it would be difficult to manage restarting each of the two processes separately?

Reply:
I'm using the queues to determine if the process has failed. So I need a separate queue per process. When they shared a queue, it was impossible to determine if one of the senders had died since the queue was kept alive by the other sender.